### ISSUE TITLE
search function - simple text

### ISSUE THREAD

**Issue by kerszl** [pwndbg/pwndbg#2455]
Greets

### Description

For some time now, the search function has not been working properly.

When I enter a search with a given string, I get an error. Searching by the string, of course ending it with zero, only finds the end of the string, not the beginning. Searching by bytes works, but it's a bit annoying

Error:
```
pwndbg> search What
Searching for value: 'What'
'search': Search memory for byte sequences, strings, pointers, and integer values.

By default search results are cached. If you want to cache all results, but only print a subset, use --trunc-out. If you want to cache only a subset of results, and print the results immediately, use --limit. The latter is specially useful if you're searching a huge section of memory.
Exception occurred: search: string argument without an encoding (<class 'TypeError'>)
```
OK:
```
pwndbg> search -x 57686174
Searching for value: b'What'
chk             0x555555556030 'What do you think the flag is?'
chk             0x555555557030 'What do you think the flag is?'
```
Ok, but I don't always expect it
```
pwndbg> search -t string What
Searching for value: b'What\x00'
```
### My setup

Gdb:      12.1
Python:   3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0]
Pwndbg:   2024.08.29 build: c2bf902e
Capstone: 5.0.1280
Unicorn:  2.0.1

**Comment by disconnect3d** [comment]
cc: @mbrla0 I bet this bug was introduced due to lldb port refactorings? :/

**Comment by mbrla0** [comment]
> cc: @mbrla0 I bet this bug was introduced due to lldb port refactorings? :/

This issue is the same as #2420. The LLDB refactorings caused `search -t bytes` to break, because now `pwndbg.search.search` actually enforces the `bytes` typing in its signature. We have to fix this in the search command itself, but I'm not familiar enough with what `search -t bytes` is supposed to do to fix it myself without changing its semantics.

Like, we could just do something like `value.encode("ascii")` in `pwndbg/commands/search.py` if we're dealing with `-t binary`, and that could be a fairly quick fix. I could do that later, if you want, and think that's good enough for a solution.

**Comment by mbrla0** [comment]
> Like, we could just do something like `value.encode("ascii")` in `pwndbg/commands/search.py` if we're dealing with `-t binary`, and that could be a fairly quick fix. I could do that later, if you want, and think that's good enough for a solution.

Turns out figuring out the old behavior exactly was much more straightforward than I thought, and we can just emulate it wholesale. I've put the fix in a PR.

**Comment by disconnect3d** [comment]
Fixed in #2458
